File: client/src/app/modules/quotation/model/IQuotation.ts
import { Voucher } from "../../crm/model/Voucher";
import { CustomerOrderOrigin } from "../../miscellaneous/model/CustomerOrderOrigin";
import { IAttachment } from "../../miscellaneous/model/IAttachment";
import { IDocument } from "../../miscellaneous/model/IDocument";
import { QuotationAbandonReason } from "../../miscellaneous/model/QuotationAbandonReason";
import { SpecialOffer } from "../../miscellaneous/model/SpecialOffer";
import { Responsable } from "../../tiers/model/Responsable";
import { AssoAffaireOrder } from "./AssoAffaireOrder";
import { CustomerOrderComment } from "./CustomerOrderComment";

export interface IQuotation extends IAttachment, IDocument {
  id: number;
  validationId: number;
  responsable: Responsable | undefined;
  specialOffers: SpecialOffer[] | undefined;
  createdDate: Date;
  description: string;
  isQuotation: boolean;
  assoAffaireOrders: AssoAffaireOrder[];
  firstReminderDateTime: Date;
  secondReminderDateTime: Date;
  thirdReminderDateTime: Date;
  customerOrderOrigin: CustomerOrderOrigin;
  abandonReason: QuotationAbandonReason;
  customerOrderComments: CustomerOrderComment[];
  lastStatusUpdate: Date;
  affairesList: string;
  servicesList: string;
  isHasNotifications: boolean;
  voucher: Voucher;
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/model/IQuotation.java
package com.jss.osiris.modules.osiris.quotation.model;

import java.time.LocalDateTime;
import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.Attachment;
import com.jss.osiris.modules.osiris.miscellaneous.model.CustomerOrderOrigin;
import com.jss.osiris.modules.osiris.miscellaneous.model.Document;
import com.jss.osiris.modules.osiris.miscellaneous.model.IAttachment;
import com.jss.osiris.modules.osiris.miscellaneous.model.IDocument;
import com.jss.osiris.modules.osiris.miscellaneous.model.IId;
import com.jss.osiris.modules.osiris.miscellaneous.model.SpecialOffer;
import com.jss.osiris.modules.osiris.profile.model.Employee;
import com.jss.osiris.modules.osiris.tiers.model.Responsable;

public interface IQuotation extends IId, IAttachment, IDocument {
	public Integer getId();

	public void setId(Integer id);

	public Employee getAssignedTo();

	public void setAssignedTo(Employee assignedTo);

	public Responsable getResponsable();

	public void setResponsable(Responsable responsable);

	public List<SpecialOffer> getSpecialOffers();

	public void setSpecialOffers(List<SpecialOffer> specialOffers);

	public LocalDateTime getCreatedDate();

	public void setCreatedDate(LocalDateTime createdDate);

	public String getDescription();

	public void setDescription(String description);

	public List<Attachment> getAttachments();

	public void setAttachments(List<Attachment> attachments);

	public List<Document> getDocuments();

	public void setDocuments(List<Document> documents);

	public List<AssoAffaireOrder> getAssoAffaireOrders();

	public void setAssoAffaireOrders(List<AssoAffaireOrder> assoAffaireOrder);

	public Boolean getIsQuotation();

	public void setIsQuotation(Boolean isQuotation);

	public CustomerOrderOrigin getCustomerOrderOrigin();

	public void setCustomerOrderOrigin(CustomerOrderOrigin customerOrderOrigin);

	public QuotationAbandonReason getAbandonReason();

	public void setAbandonReason(QuotationAbandonReason abandonReason);

}
